decoder/stats.go

```diff
@@ -375,6 +422,156 @@ func updatePokemonStats(old *Pokemon, new *Pokemon, areas []geo.AreaName) {
 	}
 }
 
+func updateRaidStats(old *Gym, new *Gym, areas []geo.AreaName) {
+	if len(areas) == 0 {
+		areas = []geo.AreaName{
+			{
+				Parent: "world",
+				Name:   "world",
+			},
+		}
+	}
+
+	locked := false
+
+	// Loop though all areas
+	for i := 0; i < len(areas); i++ {
+		area := areas[i]
+
+		// Raid has started/is active.
+		if new.RaidPokemonId.ValueOrZero() > 0 {
```

Comment:
```suggestion
		if new.RaidPokemonId.ValueOrZero() > 0 && (old == nil || old.RaidPokemonId != new.RaidPokemonId || old.RaidEndTimestamp != new.RaidEndTimestamp) {
```

that could really make it... guess it's the best check

Reply:
Sorry about the delay. Yeah, have added it now and removed the second if statement.
So hopefully it should be perfect now 😄